### DISCUSSION TITLE
Why don't my network requests appear in the Firebase Performance console?

### DISCUSSION THREAD

**Discussion by gcarpi** [capawesome-team/capacitor-firebase#755] [Q&A] [answered]
Hi everyone, how’s it going?

I could use some help with integrating Firebase Performance. I added the `@capacitor-firebase/performance` module to my app and completed all the necessary configurations as in the documentation.

The integration appears to be successful since I can see custom traces like `_app_start`, `_app_in_foreground`, etc. However, on the Network Requests tab, the only entries I’m seeing are `cdn.branch.io/**` for Android, and `firebase-settings.crashlytics.com/**`, `fcmtoken.googleapis.com/**`, `app-analytics-services.com/**`, `firebaseremoteconfig.googleapis.com/**` for iOS. Where are the rest of the network requests?

I searched for help online and found that some requests may not be captured automatically. Could this be the case here? If so, I noticed there’s a method called `newHttpMetric` in the original Firebase Performance module for manually registering requests, but I couldn’t find this method in the Capacitor library. For context, I’m using Axios on the client side to make API requests in my app.

**Libraries in use:**
- @capacitor/app@6.0.0
- @capacitor-firebase/performance@6.2.0
- @ionic/react@8.2.6
- axios@1.7.3

Thank you in advance!

![iOS](https://github.com/user-attachments/assets/a761baa2-069e-4c93-965f-10e68cd5af96)
![Android](https://github.com/user-attachments/assets/6abda601-c0d0-41da-98e2-a923d8e18c3d)

**Comment by robingenz** [comment, marked as answer]
Thank you for creating this discussion! The Firebase Performance SDK can only capture http requests on the native layer. You can try to use the [Capacitor HTTP](https://capacitorjs.com/docs/apis/http) plugin but i have not tested this myself. See https://firebase.google.com/docs/perf-mon/network-traces for more information.

**Reply by gcarpi** [reply]
Hi @robingenz, thank you for the quick answer.

Got it! I'm wondering if the simplest thing for me now would be to install and configure the web version of Firebase Performance, so I can see the requests in another project in the Firebase console. Changing the HTTP Request library from Axios to this one from Capacitor would be a big change for now.

Thanks again.

**Reply by gcarpi** [reply]
Hi @robingenz, do you think we could add the [newHttpMetric](https://firebase.google.com/docs/reference/kotlin/com/google/firebase/perf/metrics/HttpMetric) method to this module? I think it could help us manually track HTTP requests on the web side.

**Reply by robingenz** [reply]
Sure, feel free to create a feature request.
